components/raftstore/src/store/worker/region.rs

```diff
@@ -923,11 +952,19 @@ where
                 start_key,
                 end_key,
             } => {
-                fail_point!("on_region_worker_destroy", true, |_| {});
-                // try to delay the range deletion because
-                // there might be a coprocessor request related to this range
-                self.insert_pending_delete_range(region_id, start_key, end_key);
-                self.clean_stale_ranges();
+                let region_cleaner = self.region_cleaner.clone();
+                self.region_cleanup_pool
+                    .spawn(async move {
+                        fail_point!("on_region_worker_destroy", region_id == 1000, |_| {});
+                        let mut region_cleaner = region_cleaner.lock().unwrap();
+                        // try to delay the range deletion because
+                        // there might be a coprocessor request related to this range
+                        region_cleaner.insert_pending_delete_range(region_id, start_key, end_key);
+                        region_cleaner.clean_stale_ranges();
```

Comment:
It looks like they are executed asynchronously, which may cause problems with accidentally deleting the just applied snapshot.